tests/core/test_exp_manager.py

```diff
@@ -0,0 +1,109 @@
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"explicit_log_dir": str(tmp_path)})
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"resume": True})
+
+        # Check that exp_manager uses trainer.logger
+        log_dir = exp_manager(test_trainer, {"create_tensorboard_logger": False, "create_checkpoint_callback": False})
+        assert log_dir.resolve() == Path("./lightning_logs/0").resolve()
+        assert Path("./lightning_logs").exists()
+        assert Path("./lightning_logs/0").exists()
+
+    @pytest.mark.unit
+    def test_checkpoint_configurations(self, cleanup_local_folder):
+        """ Test that trainer creating modelcheckpoint and asking exp_manager to do it too results in errors, but
+        is error free if only one is asked to do so.
+        """
+        disable_tb_logger = {"create_tensorboard_logger": False}
+        test_trainer = pl.Trainer()  # Should create logger and modelcheckpoint
+        with pytest.raises(CheckpointMisconfigurationError):  # Fails because both try to create modelcheckpoint
+            exp_manager(test_trainer, disable_tb_logger)
+
+        # Should success without error
+        exp_manager(test_trainer, {"create_checkpoint_callback": False}.update(disable_tb_logger))
+
+        test_trainer_2 = pl.Trainer(checkpoint_callback=False)
+        exp_manager(test_trainer_2, disable_tb_logger)  # Should success without error
+
+    @pytest.mark.unit
+    def test_default_log_dir(self, cleanup_local_folder):
+        """Check the default of ./NeMo_experiments/default/datetime works as intended"""
+        test_trainer = pl.Trainer(checkpoint_callback=False, logger=False)
+
+        log_dir = exp_manager(test_trainer, {"create_tensorboard_logger": False, "create_checkpoint_callback": False})
+        assert (log_dir / "..").resolve() == Path("./NeMo_experiments/default/").resolve()
+        assert Path("./NeMo_experiments").exists()
+        assert Path("./NeMo_experiments/default/").exists()
+        sub_dirs = [x for x in Path("./NeMo_experiments/default/").iterdir() if x.is_dir()]
+        assert len(sub_dirs) == 1
+        assert re.match(r"[0-9]{4}-[0-9]{2}-[0-9]{2}_[0-9]{2}-[0-9]{2}-[0-9]{2}", sub_dirs[0].name)
+
+    @pytest.mark.unit
+    def test_log_dir_overrides(self, tmp_path):
+        """Check a variety of trainer options with exp_manager"""
+        test_trainer = pl.Trainer(checkpoint_callback=False, logger=False)
+
+        log_dir = exp_manager(test_trainer, {"explicit_log_dir": str(tmp_path / "test_log_dir_overrides")})
+        assert log_dir.resolve() == (tmp_path / "test_log_dir_overrides").resolve()
+        assert Path(tmp_path).exists()
+        assert Path(tmp_path / "test_log_dir_overrides").exists()
+
+    # Test log dir creation
+    # Test logging
+    # Test find_last_checkpoint
```

Comment:
Im interested to know how pytorch lightning handles restoring the state of the scheduler - specifically its current iteration number (`last_epoch`). This should be a part of the tests (in a future PR if necessary).

Reply:
I don't have any plans to test trainer.resume_from_checkpoint inside of test_exp_manager. Mostly because I can't save a model checkpoint because I don't have a model.

This is a good test to have though, although at a higher level than a unittest.